drv/transceivers-server/src/net_api.rs

```diff
@@ -0,0 +1,410 @@
+                if data_size as usize > data.len() {
+                    return Err(Error::RequestTooLarge);
+                }
+                self.write(mem, modules, data)?;
+                Ok((HostResponse::Write(mem), 0))
+            }
+            HostRequest::SetPowerMode(mode) => {
+                let mask = get_mask(modules)?;
+                // TODO: do we need delays in between any of these operations?
+                match mode {
+                    PowerMode::Off => {
+                        // Power disabled, LpMode enabled (the latter doesn't
+                        // make a difference, but keeps us in a known state)
+                        self.transceivers
+                            .clear_power_enable(mask)
+                            .map_err(|_e| Error::ReadFailed)?;
+                        self.transceivers
+                            .set_reset(mask)
+                            .map_err(|_e| Error::ReadFailed)?;
+                        self.transceivers
+                            .set_lpmode(mask)
+                            .map_err(|_e| Error::ReadFailed)?;
+                    }
+                    PowerMode::Low => {
+                        // Power enabled, LpMode enabled
+                        self.transceivers
+                            .set_lpmode(mask)
+                            .map_err(|_e| Error::ReadFailed)?;
+                        self.transceivers
+                            .set_power_enable(mask)
+                            .map_err(|_e| Error::ReadFailed)?;
+                        self.transceivers
+                            .clear_reset(mask)
+                            .map_err(|_e| Error::ReadFailed)?;
+                    }
+                    PowerMode::High => {
+                        // Power enabled, LpMode disabled
+                        self.transceivers
+                            .clear_lpmode(mask)
+                            .map_err(|_e| Error::ReadFailed)?;
+                        self.transceivers
+                            .set_power_enable(mask)
+                            .map_err(|_e| Error::ReadFailed)?;
+                    }
+                }
+                Ok((HostResponse::Ack, 0))
+            }
+            HostRequest::ManagementInterface(i) => {
+                todo!()
+            }
+        }
+    }
+
+    fn select_page(
+        &mut self,
+        upper_page: UpperPage,
+        modules: ModuleId,
+    ) -> Result<(), Error> {
+        let mask = get_mask(modules)?;
+        match upper_page {
+            UpperPage::Cmis(page) => {
+                self.transceivers
+                    .set_i2c_write_buffer(&[page.page()])
+                    .map_err(|_e| Error::ReadFailed)?;
+                self.transceivers
+                    .setup_i2c_write(0x7F, 1, mask)
+                    .map_err(|_e| Error::ReadFailed)?;
+                sleep_for(5); // TODO: Poll instead
```

Comment:
As of https://github.com/oxidecomputer/hubris/commit/ae111d6b5a027c1c987cc7ce8e8eb5c663cb82ca, you can now poll busy registers.

Reply:
Done in the recent rebase!